### DISCUSSION TITLE
make it possible to initialize the configuration in a parallel way

### DISCUSSION THREAD

**Discussion by yepeekai** [LuckyPennySoftware/AutoMapper#3639] [Ideas]
new MapperConfiguration takes a couple of seconds in our case... would it be possible to make it happen in a parallel way?  As a configuration option... I don't care much about the server but on our dev machines, we have to launch the app a lot of times per day and automapper initialization is slowing things down.  I have a 12 cores machines so I think it would be nice to speed things up using parallelism on dev machines.

**Comment by lbargaoanu** [comment]
Try the [MyGet](https://docs.automapper.org/en/latest/The-MyGet-build.html) build. And see [this](https://github.com/AutoMapper/AutoMapper/blob/5a9e5fd1c72bf50acafda2cf0f35424261f78bd0/docs/11.0-Upgrade-Guide.md#configuration-performance). The parallel stuff you have to implement yourself by having multiple `MapperConfiguration` objects. But, before going that route, you shoud profile it and find out where exactly that time is spent. Surely there are opportunities to make it faster.
